chainer/gradient_check.py

```diff
@@ -131,20 +131,22 @@ def check_backward(func, x_data, y_grad, params=(),
     :func:`numerical_grad` to calculate numerically the gradients and compares
     the types of gradients with :func:`chainer.testing.assert_allclose`.
 
-    To reduce computational time, it uses a function
+    To reduce computational time, it uses directional derivative along a
+    rando vector. A function
     :math:`g: \\mathbb{R} \\rightarrow \\mathbb{R}^n` defined as
-    :math:`g(\\alpha) = f(\\alpha x)`, where :math:`\\alpha \\in \\mathbb{R}`
-    and :math:`f` is a function which actually
-    you want to test.
+    :math:`g(\\delta) = f(x + \\delta r)`, where
+    :math:`\\delta \\in \\mathbb{R}`, :math:`r \\in \\mathbb{R}^n`
+    is a random vector
+    and :math:`f` is a function which actually you want to test.
     Its gradient is
 
     .. math::
-       g'(\\alpha) = f'(\\alpha x) \\cdot x.
+       g'(\\delta) = f'(x + \\delta r) \\cdot r.
 
-    When :math:`\\alpha = 1`, :math:`g'(1) = f'(x) \\cdot x`.
-    So :math:`g'(1)` is calculated with :func:`numerical_grad` and
+    When :math:`\\delta = 0`, :math:`g'(0) = f'(x) \\cdot r`.
+    So :math:`g'(0)` is calculated with :func:`numerical_grad` and
     compared with dot product of the gradient :math:`f` and
```

Comment:
Suggestion of explanation:

So we can check the correctness of back propagation of `f` indirectly by comparing this equation with the gradient of `g` numerically calculated and that of `f` computed with backprop. If `r` is chosen from uniform distribution, we can conclude with high probability that the gradient of `f` itself is correct.